projects/bp-gallery/src/components/views/geoguessr/GeoView.tsx

```diff
@@ -0,0 +1,149 @@
+import { Box, Button, Modal, Typography } from '@mui/material';
+import { Info } from '@mui/icons-material';
+import { useEffect, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  useGetAllPicturesByArchiveQuery,
+  useGetPictureGeoInfoQuery,
+} from '../../../graphql/APIConnector';
+import { useSimplifiedQueryResponseData } from '../../../graphql/queryUtils';
+import { FlatArchiveTag } from '../../../types/additionalFlatTypes';
+import ZoomWrapper from '../picture/overlay/ZoomWrapper';
+import GeoMap from './GeoMap';
+import { useStorageState } from 'react-use-storage-state';
+import useGetPictureLink from '../../../hooks/get-pictureLink.hook';
+
+const getAllPictureIds = (archives: FlatArchiveTag[]) => {
+  const list: string[] = [];
+  archives.forEach(archive => archive.pictures?.forEach(picture => list.push(picture.id)));
+  return list;
+};
+
+const shufflePictureIds = (pictureIds: string[], seed: number) => {
+  const newPictureIds = [...pictureIds];
+  for (let i = newPictureIds.length - 1; i > 0; i--) {
+    const j = Math.abs(Math.floor(i * Math.sin(i % seed)));
+    [newPictureIds[i], newPictureIds[j]] = [newPictureIds[j], newPictureIds[i]];
+  }
+  return newPictureIds;
+};
+
+const getTodaysPictureQueue = (pictureIds: string[]) => {
+  const pictureNumber = 10;
+  const currentDate = new Date();
+  const startDate = new Date(currentDate.getFullYear(), 0, 1);
+  const days = Math.floor((currentDate.valueOf() - startDate.valueOf()) / (24 * 60 * 60 * 1000));
+  const resultIndex = pictureIds.length % days;
+  const list = [];
+  for (let i = 0; i < pictureNumber; i++) {
+    list.push(pictureIds[(resultIndex + i) % pictureIds.length]);
+  }
+  return list;
+};
+
+const GeoView = () => {
+  const { t } = useTranslation();
+  const fallbackPictureId = '3';
+  const pictureQueue = useRef(['']);
+  const seed = 42;
+
+  const [hasReadInstructions, setHasReadInstructions] = useStorageState<boolean>(
+    'hasReadInstructions',
+    false
+  );
+  const [modalOpen, setModalOpen] = useState(!hasReadInstructions);
+  const dontShowAgain = () => {
+    setHasReadInstructions(true);
+    setModalOpen(false);
+  };
+
+  const { data: picturesData } = useGetAllPicturesByArchiveQuery();
+  const archives: FlatArchiveTag[] | undefined =
+    useSimplifiedQueryResponseData(picturesData)?.archiveTags;
+
+  useEffect(() => {
+    const allPictureIds = archives ? getAllPictureIds(archives) : [];
```

Comment:
I think you might want to return instead of continuing to work with `[]`, since you'll initialize `pictureQueue.current` with an Array of `undefined`s otherwise. I get an error that `$pictureId` is not provided when starting GeoGuesser, which I think has its origin here.